src/main/java/org/wso2/securevault/secret/repository/VaultSecretRepositoryProvider.java

```diff
@@ -0,0 +1,152 @@
+import org.wso2.securevault.commons.MiscellaneousUtil;
+import org.wso2.securevault.keystore.IdentityKeyStoreWrapper;
+import org.wso2.securevault.keystore.TrustKeyStoreWrapper;
+import org.wso2.securevault.secret.SecretRepository;
+import org.wso2.securevault.secret.SecretRepositoryProvider;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * This class is responsible for initializing SecretRepositories which belongs to a particular provider.
+ * This is being called by the SecretManager by providing the secret configuration properties and the provider type
+ * and this will return a collection of initialized SecretRepositories.
+ *
+ * @see org.wso2.securevault.secret.SecretRepository
+ * @see org.wso2.securevault.secret.SecretManager
+ */
+public class VaultSecretRepositoryProvider implements SecretRepositoryProvider {
+
+    private static final Log log = LogFactory.getLog(VaultSecretRepositoryProvider.class);
+
+    // Property String for secretProviders.
+    private final static String PROP_SECRET_PROVIDERS = "secretProviders";
+
+    // Property String for repositories.
+    private final static String PROP_REPOSITORIES = "repositories";
+
+    // Property String for properties.
+    private final static String PROPERTIES = "properties";
+
+    // Dot String.
+    private final static String DOT = ".";
+
+    // Contains all initialized secret repositories under provider type vault.
+    private final Map<String, SecretRepository> vaultRepositoryMap = new HashMap<>();
+
+    /**
+     * @see org.wso2.securevault.secret.SecretRepositoryProvider
+     */
+    @Override
+    public SecretRepository getSecretRepository(IdentityKeyStoreWrapper identity, TrustKeyStoreWrapper trust) {
+
+        return null;
+    }
+
+    /**
+     * Returns a map containing initialized secret repositories corresponds to a give provider type.
+     *
+     * @param configurationProperties All the properties under secret configuration file.
+     * @param providerType            Type of the VaultSecretRepositoryProvider class.
+     * @return Initialized secret repository map.
+     * @throws SecureVaultException when creating the SecretRepository instances.
+     */
+    @Override
+    public Map<String, SecretRepository> initProvider(Properties configurationProperties, String providerType)
+            throws SecureVaultException {
+
+        // Get the list of repositories from the secret configurations.
+        StringBuilder repositoriesStringPropKey = new StringBuilder()
+                .append(PROP_SECRET_PROVIDERS)
+                .append(DOT)
+                .append(providerType)
+                .append(DOT)
+                .append(PROP_REPOSITORIES);
+
+        String repositoriesString = MiscellaneousUtil.getProperty(configurationProperties,
+                repositoriesStringPropKey.toString(), null);
+
+        if (MiscellaneousUtil.isValidPropertyValue(repositoriesString)) {
```

Comment:
Why do we need to define this method in MiscellaneousUtil class ?

Reply:
because this is being used by the SecretsManager as well to validate the property value